fellow-submissions/stephanie-smith.md

```diff
@@ -0,0 +1,108 @@
+
+**2. What will the following code log? Why?**
+    ```javascript
+    let a = 'outer';
+
+    function testScope() {
+      let a = 'inner';
+      console.log(a);
+    }
+
+    console.log(a);
+    testScope();
+    console.log(a);
+    ```
+  - It will first log 'outer', then it will log 'inner', then it will log 'outer' again. The global scope will be called first because there is no direct reference to the scope that 'inner' is in. However, when the function is called, `a`'s value changes into what it is in the scope, and logs that out instead. Then it refers back to the global scope.
+
+**3. What will the following code log? Why?**
+    ```javascript
+    function hello() {
+      let a = 'hello';
+    }
+
+    hello();
+    console.log(a);
+    ```
+  - It will log 'outer', then 'a is not defined' , because the function `hello()` was called first, allowing the variable `a` to log its value. However, when you console.log `a` without calling the function, it does not work because there is no `a` in the global scope. Since the `a` in the function can't be accessed from the global scope, it simply remains undefined because `a` technically does not exist.
+
+**4. What does the following code log? Why?**
+    ```javascript
+    console.log(a);
+
+    var a = 1;
+    ```
+  - It will log undefined because `a` was declared, but not initialized. It then calls upon `a`, which is still undefined, and only after it's called is it defined to be equal to 1.
+  
+**5. What does the following code log? Why?**
+    ```javascript
+    console.log(a);
+
+    let a = 1;
+    ```
+  - This will log an error because `a` is not defined. `let` variables do not get initialized with anything, so there will be a reference error.
+
+**6. In the code snippet below, the function declaration comes _after_ the variable declaration in the code below. Why does it still work?**
+    ```javascript
+    let name = "Devonte";
+
+    function sayItLoud() {
+      console.log(`${name}!!!`)
+    }
+
+    sayItLoud();
+    ```
+  - Local scopes can access variables that are declared and initialized in the global scope, but global scopes can't access variables in the local scope, so in this case, it will still work.
+    
+**7. Why does the following code throw an error? What is the key difference between it and the example above?**
+    ```javascript
+    sayItLoud();
+
+    function sayItLoud() {
+      console.log(`${greeting}!!!}`)
+    }
+
+    let greeting = "Hello"
+    ```
+  - The function is called before it is even declared, so `greeting` doesn't even exist yet despite it being in the global scope.
+
+**8. What does the following code log? Why?**
+    ```javascript
+    let problemSet;
+
+    function complete(assignment) {
+      assignment = "done";
+    }
+
+    complete(problemSet);
+    console.log(`Status for Problem Set: ${problemSet}`);
+    ```
+  - It logs "Status for Problem Set: undefined" because problemSet was never initialized, so it becomes undefined.
```

Comment:
Yes. But also, tell me why the assignment on line 4 (inside of the `complete` function) does not affect the variable `problemSet` defined on line 1.